src/Ledger/PParams.lagda

```diff
@@ -38,43 +41,89 @@ record PoolThresholds : Set where
+\begin{code}
+        Emax               : Epoch
+        collateralPercent  : ℕ
+\end{code}
+\emph{Governance group}
+\AgdaTarget{drepThresholds, poolThresholds, minCCSize, ccTermLimit, govExpiration, govDeposit, drepDeposit, drepActivity, minimumAVS}
+\begin{code}
+        drepThresholds     : DrepThresholds
+        poolThresholds     : PoolThresholds
+        minCCSize          : ℕ
+        ccTermLimit        : ℕ
+        govExpiration      : ℕ
+        govDeposit         : Coin
+        drepDeposit        : Coin
+        drepActivity       : Epoch
+        minimumAVS         : Coin
 
 paramsWellFormed : PParams → Bool
 paramsWellFormed pp = ⌊ ¬? (0 ∈? setFromList
   (maxBlockSize ∷ maxTxSize ∷ maxHeaderSize ∷ maxValSize ∷ minUTxOValue ∷ poolDeposit
   ∷ collateralPercent ∷ ccTermLimit ∷ govExpiration ∷ govDeposit ∷ drepDeposit ∷ [])) ⌋
   where open PParams pp
 \end{code}
-\caption{Definitions used for protocol parameters}
-\label{fig:defs:pparams}
+\end{AgdaAlign}
+} %% End: small
+\caption{Protocol parameter declarations}
+\label{fig:protocol-parameter-declarations}
 \end{figure*}
+The \AgdaDatatype{ProtVer} type represents the protocol version used in the Cardano ledger.
+It is a pair of natural numbers, the first of which represents the major version, the second
+represents the minor version.
+
+The \AgdaRecord{PParam} type is a record containing parameters used in the Cardano ledger, which we group according
+to the general purpose that each parameter serves. This allows thresholds to be set by group.
+\begin{itemize}
+  \item \AgdaInductiveConstructor{NetworkGroup}: parameters related to the network settings;
+  \item \AgdaInductiveConstructor{EconomicGroup}: parameters related to the economic aspects of the ledger;
+  \item \AgdaInductiveConstructor{TechnicalGroup}: parameters related to technical settings;
+  \item \AgdaInductiveConstructor{GovernanceGroup}: parameters related to governance settings.
+\end{itemize}
+We do not require that each protocol parameter governance action be confined to a single group.
+In case a governance action carries updates for multiple parameters from different groups,
+the maximum threshold of all the groups involved will apply to any given such governance action.
+
+The \emph{Network}, \emph{Economic}, and \emph{Technical} parameter groups contain protocol parameters that
+were already introduced during the Shelley, Alonzo and Babbage eras.
+The new protocol parameters introduced in the Conway era (CIP-1694) belong to the \emph{Governance} group.
+These new pramaters are declared in Figure~\ref{fig:protocol-parameter-declarations} and denote the following concepts.
+\begin{itemize}
+  \item \AgdaField{drepThresholds}: governance thresholds for DReps; these are rational numbers
+  drawn from the following set: \{\AgdaField{P1}, \AgdaField{P2a}, \AgdaField{P2b}, \AgdaField{P3}, \AgdaField{P4}, \AgdaField{P5a}, \AgdaField{P5b}, \AgdaField{P5c}, \AgdaField{P5d}, \AgdaField{P6}\};
```

Comment:
This makes it sound like `drepThresholds` is a subset of the set of `P`s. How about 'these are rational numbers named P1, ...'?

Reply:
Actually, I think my working is a bit better, but I'm also fine with the change you suggest... so I'll change it. 👍🏼